lib/quickPrintApi.js

```diff
@@ -1,24 +1,44 @@
 import { request } from './api';
+import { QUICKETS_SERVER_TYPE, ALBERT_SERVER_TYPE } from './constants';
 
-export default function printToPrinter(printer, page) {
+function postData(api_key, body) {
+  return  {
+            method: 'POST',
+            credentials: 'same-origin',
+            headers: {
+              'Content-Type': 'application/json',
+              'x-api-key': api_key,
+            },
+            body: JSON.stringify(body),
+          };
+}
+
+function printToPrinter(printer, page) {
   const body = {
     printer_name: printer.description,
-    printer: printer.id, // deprecated
+    printer: printer.id,             // deprecated
     api_key: printer.server.api_key, // deprecated
     tickets: page.tickets,
     page_format: page.page_format,
-    base_url: page.base_url,
+    base_url: page.base_url,         // deprecated
   };
 
-  const postData = {
-    method: 'POST',
-    credentials: 'same-origin',
-    headers: {
-      'Content-Type': 'application/json',
-      'x-api-key': printer.server.api_key,
-    },
-    body: JSON.stringify(body),
+  return request(`${printer.server.host}/print-tickets`, postData(printer.server.api_key, body));
+}
+
+function printToReceipt(printer, page) {
+  const body = {
+    printer_name: printer.description,
+    tickets: page.tickets,
   };
 
-  return request(`${printer.server.host}/print-tickets`, postData);
+  return request(`${printer.server.host}/print-receipts`, postData(printer.server.api_key, body));
+}
+
+export default function print(printer, page, printServerType) {
+  if (printServerType === QUICKETS_SERVER_TYPE) {
+    return printToPrinter(printer, page);
+  }
+
+  return printToReceipt(printer, page);
```

Comment:
I think the naming here isn't clear for me. e.g. printToReceipt itself sounds confusing because we're printing tickets and receipts in both cases (database and the thermal paper printers [which we call receipt printers]).

Maybe it's worth a think about what it is that makes QUICKETS/ALBERT types different? Is it the fact that quickets are essentially fixed size? Whereas Albert-style is basically a paper print that is of no particular length.

If there are pragmatic reasons to just merge but we acknowledge this and will fix it later, please let me know.